Mathlib/Topology/Order/Monotone.lean

```diff
@@ -28,12 +28,20 @@ variable [ConditionallyCompleteLinearOrder α] [TopologicalSpace α] [OrderTopol
 
 /-- A monotone function continuous at the supremum of a nonempty set sends this supremum to
 the supremum of the image of this set. -/
-theorem Monotone.map_sSup_of_continuousAt' {f : α → β} {A : Set α} (Cf : ContinuousAt f (sSup A))
-    (Mf : Monotone f) (A_nonemp : A.Nonempty) (A_bdd : BddAbove A := by bddDefault) :
+theorem MonotoneOn.map_sSup_of_continuousWithinAt' {f : α → β} {A : Set α}
+    (Cf : ContinuousWithinAt f A (sSup A))
+    (Mf : MonotoneOn f A) (A_nonemp : A.Nonempty) (A_bdd : BddAbove A := by bddDefault) :
     f (sSup A) = sSup (f '' A) :=
```

Comment:
For lemmas like this, stating the property of some set depending on some function property (continuity, monotonicity, convexity, differentiability...), I tend to suggest stating the required function property assumptions on a *superset* of the set in the conclusion. This is useful, because usually the set `A` you're trying to plug in is *not* the domain on which the function respect the property, which would rather be an open half-line, or the complement of a singleton or what not.

Here, concretely, I would do
```suggestion
theorem MonotoneOn.map_sSup_of_continuousWithinAt' {f : α → β} {A B : Set α} (hAB : A ⊆ B)
    (Cf : ContinuousWithinAt f B (sSup A))
    (Mf : MonotoneOn f B) (A_nonemp : A.Nonempty) (A_bdd : BddAbove A := by bddDefault) :
    f (sSup A) = sSup (f '' A) :=
```
or even
```suggestion
theorem MonotoneOn.map_sSup_of_continuousWithinAt' {f : α → β} {A B C : Set α} (hAB : A ⊆ B)
    (hAC : A ⊆ C) (Cf : ContinuousWithinAt f B (sSup A))
    (Mf : MonotoneOn f C) (A_nonemp : A.Nonempty) (A_bdd : BddAbove A := by bddDefault) :
    f (sSup A) = sSup (f '' A) :=
```
but that's probably overengineered

Reply:
Fair enough, I went with the first option.

Reply:
Actually Bhavik disagrees that this is worth doing except for things like `ConvexOn` (because `ConvexOn.subset` is so annoying to use), and I am somewhat convinced by this argument. Would you mind rolling back?

Reply:
No problem -- I confess I was a bit skeptical as well, when constructing a proof using `refine` I think it's a bit easier to deal with the subset thing afterwards.